vsphere/tests/test_check.py

```diff
@@ -81,3 +81,39 @@ def test_external_host_tags(aggregator, realtime_instance):
 
     check.set_external_tags = MagicMock()
     check.submit_external_host_tags()
+
+
+@pytest.mark.usefixtures('mock_type', 'mock_threadpool', 'mock_api')
+def test_collect_metric_instance_values(aggregator, dd_run_check, realtime_instance):
+    realtime_instance.update(
+        {
+            'collect_per_instance_filters': {
+                'vm': [r'cpu\.usage\.raw', r'disk\..*'],
+                'host': [r'cpu\.coreUtilization\..*', r'sys\.uptime\..*', r'disk\..*'],
+            }
+        }
+    )
+    check = VSphereCheck('vsphere', {}, [realtime_instance])
+    dd_run_check(check)
+
+    # Following metrics should match and have instance value tag
+    aggregator.assert_metric(
+        'vsphere.cpu.usage.raw', tags={'cpu_core:4', 'vcenter_server:FAKE'},
```

Comment:
First time I see tags being asserted as a set, :mindblown:

Reply:
My bad, it's typo. Fixed.